pkg/subnet/byteset.go

```diff
@@ -0,0 +1,101 @@
+package subnet
+
+import "fmt"
+
+// ByteSet represents 0 - 255 unique bytes.
+type ByteSet [4]uint64
+
+// Add adds a byte to this ByteSet
+func (b *ByteSet) Add(bv byte) {
+	b[bv>>6] |= uint64(1) << uint64(bv&0x3f)
+}
+
+// Remove removes a byte from this ByteSet
+func (b *ByteSet) Remove(bv byte) {
+	b[bv>>6] &^= uint64(1) << uint64(bv&0x3f)
+}
+
+// Contains returns true if this ByteSet contains the given byte
+func (b *ByteSet) Contains(bv byte) bool {
+	return b[bv>>6]&(uint64(1)<<uint64(bv&0x3f)) != 0
+}
+
+// Equals returns true if this ByteSet equals the argument
+func (b *ByteSet) Equals(other *ByteSet) bool {
+	if other == nil {
+		return false
+	}
+	return *b == *other
+}
+
+// ToSlice returns the number of bytes in this ByteSet
+func (b *ByteSet) Len() (l int) {
```

Comment:
I had no idea you could have named return variables that are automatically created and returned in functions in Go, thinking of numerous places I could have done this 🤦